docs/src/pages/docs/crocks/Writer.md

```diff
@@ -0,0 +1,352 @@
+
+```javascript
+import Assign from 'crocks/Assign'
+import Writer from 'crocks/Writer'
+
+import equals from 'crocks/pointfree/equals'
+
+// AssignWriter :: Writer Assign a
+const AssignWriter =  Writer(Assign)
+
+AssignWriter.of(2)
+  .equals(AssignWriter.of(2))
+//=> true
+
+AssignWriter(Assign({ a: 20 }), null)
+  .equals(AssignWriter(Assign({ a: 30 }), null))
+//=> true
+
+Writer(String)('here is 42', { value: 42 })
+  .equals(Writer(String)('also 42', { value: 42 }))
+//=> true
+
+Writer(String)('hello', 25)
+  .equals(Writer(String)('hello', 26))
+//=> false
+
+equals(Writer(Array).of('result'), 'result')
+//=> false
+```
+
+#### map
+
+Used to apply transformations to the value that a `Writer` holds, `map` takes a function that it will lift into the context of the `Writer` and apply to it the wrapped value. 
+
+```javascript
+import Sum from 'crocks/Sum'
+import Writer from 'crocks/Writer'
+
+// toLower :: String -> String
+const toLower = x => x.toLowerCase()
+
+Writer(Sum).of('HELLO!')
+  .map(toLower)
+//=> Writer( Sum 0 "hello!" )
+
+Writer(String)('file was loaded', 'important text from document')
+  .map(contents => contents.length)
+//=> Writer("file was loaded"  28)
+```
+
+#### ap
+
+Short for apply, `ap` is used to apply a `Writer` instance containing a value to another `Writer` instance that contains a function, resulting in new `Writer` instance comprising of the result of concatenating the monoid instances, and the value derived from applying the function. `ap` requires that it is called on a writer that wraps a curried polyadic function as its value.
+
+```javascript
+import Sum from 'crocks/Sum'
+import Writer from 'crocks/Writer'
+
+import curry from 'crocks/helpers/curry'
+import liftA2 from 'crocks/helpers/liftA2'
+
+// SumWriter :: Writer Sum a
+const SumWriter = Writer(Sum)
+
+// slash :: Writer Number String
+const slash = SumWriter(Sum(3), 'sword')
+
+// slash :: Writer Number String
+const burn = SumWriter(Sum(7), 'fireball')
+
+// doubleAttack :: NumberRec -> NumberRec -> NumberRec
+const dobuleAttack = liftA2(x => y => `You unleash a ${x} and ${y} combo`)
+
+dobuleAttack(slash, burn)
+//=> Writer( Sum 10 "You unleash a sword and fireball combo" )
+
+const LogWriter = Writer(Array)
+
+// add :: Number -> Number -> Number
+const add = curry((x, y) => x + y)
+
+// addPercentage :: Number -> Number -> Number
+const addPercentage = curry((percentage, amount) => amount + percentage * amount)
+
+// includeVatWithLog :: Number -> Writer [Number]
+const includeVatWithLog = (amount) => LogWriter([ 'adding vat' ], addPercentage(0.19, amount))
+
+// includeVatWithLog :: Number -> Writer [Number]
+const includeExpressShippingWithLog = (amount) => LogWriter([ 'adding express shipping' ], add(3.99, amount))
```

Comment:
I don't personally mind these on one line but for consistencies sake should we mulitline them? maybe @evilsoft can weigh in here. i don't want to continue something just because "that's how we've always done it" but i also think the multi-line makes it easier for a FP newbie to follow

Reply:
I really don't intend to be blunt, but it's really unproductive and demoralizing to have comments in most PRs about line lengths, how a function looks and the like. 

I wouldn't mind fixing these for this PR, but if it's really a project level constraint, it would be great to have this enforced by lint rules, or not at all.

Reply:
I agree, i hate it myself. That's why i loved it when i seen you making the new rule. I would prefer that the main focus of a PR is about the value add not the formatting of it

Reply:
Guys, love the whole Crocks library (using Async in production a lot).

I noticed that on line 175 the signature has not been updated to match the following function (// includeVatWithLog ... vs //includeExpressShippingWithLog).

Also on line 159 and following I think I spotted a typo... "dobuleAttack" should be "doubleAttack" yes? 

Thanks! Cannot wait to add Writer to the toolbox.

Reply:
Yeah, I feel you, we need to figure out a way to communicate things.
To avoid this, until we can come up with a way to communicate/enforce these styles/formatting bit. I can go in after and make those changes. things like the multiline and using `()` around single item. That feels a little shifty to me though, as if I am going back and redoing work, so I am open to other suggestions.

The importance of it, is the styling is 🌽sistant between docs. Or as 🌽sistant as we can get it. 
The reason for the multi line definition is for mobile and tablet views, it sucks to have to scroll over to see the definition and then go back and forth to do the composition in your head. 

So there are reasons for them, but you are right, without some way to define that rule in eslint, it can make these demoralizing.

Reply:
But there is value in the formatting, especially with documentation.
How information is presented in documentation does make an impact on the cognitive load required of the Learner and to account for a11y and responsive 🌽cerns.

In normal code I would say that could be debated, but for documentation it is really important to think about things like screen width, spacing between expressions, not adding unnecessary characters (like () on single arguments), or the like. This is mostly a failing on my part for not having this a documented enough.

Reply:
@fabfianda - Thanks for the corrections! Definitely missed that.